share/picongpu/tests/Synchrotron_plugin/lib/validate.py

```diff
@@ -38,57 +31,84 @@ def read_setup(folderName):
 iterNo = 4000  # iteration from which we read the data must be the same as in ../bin/validate.ci
 
 
-def main(dataPath):
-    print(f"Validating data in {dataPath}")
-    gamma, Heff, dt = read_setup(dataPath)
-
-    series = io.Series(dataPath + "/openPMD/simData_test_%T.bp", io.Access.read_only)
+def read_photon_data(series):
     it = series.iterations[iterNo]
-
-    simulation_dt = it.get_attribute("dt") * it.get_attribute("unit_time")
-    if abs(simulation_dt - dt) > 1e-10:  # check if the simulation dt is the same as the expected dt. 1e-10 is arbitrary
-        raise SystemExit(f"Simulation dt is {simulation_dt} but expected dt is {dt}\n Test failed")
-
     h = it.particles["y"]  # photons
-    e = it.particles["e"]  # electrons
 
-    # photons
+    # Reading photon weights
     h1 = h["weighting"][io.Mesh_Record_Component.SCALAR]
     w = h1[:]
     w_SI = h1.unit_SI
     series.flush()
     w *= w_SI
+
+    # Reading photon momentum
     h1 = h["momentum"]["y"]
     p_y = h1[:]
     p_y_SI = h1.unit_SI
     series.flush()
     p_y = np.float64(p_y)
     p_y *= p_y_SI / w
-    hist_data = np.abs(p_y)
+
+    # Calculate photon energy
+    hist_data = np.abs(p_y * const.c / const.elementary_charge)
+
     if len(hist_data) < 5e5:
         raise SystemExit(
             f"Number of photons is {len(hist_data)} but expected number of photons to be at least 5e5\n Test failed"
         )
-    hist_data = np.abs(p_y * const.c / const.elementary_charge)  # calculate the energy of photons
 
-    # electron energy: take electron momentum and calculate energy
+    return hist_data
+
+
+def read_electron_data(series):
+    it = series.iterations[iterNo]
+    e = it.particles["e"]  # electrons
+
+    # Reading electron weights
     h1 = e["weighting"][io.Mesh_Record_Component.SCALAR]
     e_w = h1[:]  # get the weighting
     e_w_SI = h1.unit_SI  # get the weighting unit
     series.flush()
     e_w *= e_w_SI  # convert to SI units
+
+    # Reading electron momentum
     h1 = e["momentum"]["y"]
```

Comment:
Just to be extra sure: The `"y"` in this line has nothing to do with the `"y"` that approximates a $\gamma$ and stands for photons further up, right? It's probably the direction of the momentum?

Reply:
yes, you are right

Reply:
Yes